extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/MultiStageQueryContext.java

```diff
@@ -121,6 +125,11 @@ public class MultiStageQueryContext
   public static final String CTX_INDEX_SPEC = "indexSpec";
 
   public static final String CTX_USE_AUTO_SCHEMAS = "useAutoColumnSchemas";
+  public static final boolean DEFAULT_USE_AUTO_SCHEMAS = false;
+
+  public static final String CTX_ARRAY_INGEST_MODE = "arrayIngestMode";
+  public static final ArrayIngestMode DEFAULT_ARRAY_INGEST_MODE = ArrayIngestMode.NONE;
```

Comment:
The default mode should be MVD IMHO since it will not break stuff.

Reply:
I personally disagree, I think we should default to none, then people explicitly choose to use MVDs or arrays

Reply:
mostly because the behavior of MVD is totally incorrect

Reply:
I am working on updating the docs/examples refer to the parameter and to separate them into ones that explicitly store MVDs using ARRAY_TO_MV function, and ones which only use storing array typed columns instead.

MVD mode is not going to have examples, because it should be the first to go and no one should rely on this behavior, because again the behavior is incorrect.

Reply:
IMHO we cannot break user's ingestion sql query so I propose the mode should be MVD.
The patch should also throw a warning from the controller and throw it on the console if a row signature with Array is detected. The warning clearly lays out the path of migration.

There are lot of layers build out in organizations. A change even if its adding a context flag to sometimes takes weeks to reach to production. With this warning we effectively nudge the user that we are going to break compatibility soon.